haskell-language-server.cabal

```diff
@@ -497,6 +497,9 @@ executable haskell-language-server
 
   default-language: Haskell2010
   default-extensions: DataKinds, TypeOperators
+  
+  if impl(ghc >= 9.4)
+    build-depends: ghc-exactprint >= 1.6.1
```

Comment:
why is this needed?

Reply:
Ah, a good question. Without this seemingly unnecessary constraint, Cabal somehow manages to pull `ghc-exactprint-1.6.0` - presumably because there's a dependency of dependency that's asking for it. And CI tests kept failing.

With this addition, all the relevant tests pass.

Reply:
I really think this needs to go in the right place. It shouldn't be _that_ hard to figure out. At the _very_ least it needs a clear comment explaining this.

Reply:
It is not necessary, afaict, this worked for me out of the box:
```
> cabal update
> cabal build -w ghc-9.4.3 exe:haskell-language-server
```

Reply:
Interesting - perhaps, `index-state` changed since...? Should I update `index-state` in this PR too?

Also, have you tried just `cabal build -w ghc-9.4.3` (to build whatever's there)?

Reply:
Since you updated the index-state in cabal.project, you locally need to make sure your global index-state is newer than the one in cabal.project (which you added). So just running `cabal update`  in the project or outside of it should do the trick.

Yeah, works just the same. It'd be weird if cabal didn't choose the most recent version of 'ghc-exactprint'.